sympy/matrices/matrices.py

```diff
@@ -1164,7 +1153,7 @@ def eigenvals(self, error_when_incomplete=True, **flags):
                     eigs[diagonal_entry] += 1
         else:
             flags.pop('simplify', None)  # pop unsupported flag
-            eigs = roots(mat.charpoly(x=Dummy('x'), simplify=simpfunc_1), **flags)
+            eigs = roots(mat.charpoly(x=Dummy('x'), simplify=simplify if isinstance(simplify, FunctionType) else _simplify), **flags)
```

Comment:
I would prefer having

    if isinstance(simplify, FunctionType):
        eigs = roots(mat.charpoly(x=Dummy('x'), simplify=simplify), **flags)
    else:
        eigs = roots(mat.charpoly(x=Dummy('x'), **flags)

instead of one long line.